File: frontend/src/app/shared/dto/tmdb/TmdbSeriesDetailDto.ts
import {TmdbCreatorDto} from "./TmdbCreatorDto";
import {TmdbEpisodeDto} from "./TmdbEpisodeDto";
import {TmdbGenreDto} from "./TmdbGenreDto";
import {TmdbNetworkDto} from "./TmdbNetworkDto";
import {TmdbSeasonDto} from "./TmdbSeasonDto";
import {TmdbProductionCompanyDto} from "./TmdbProductionCompanyDto";

export interface TmdbSeriesDetailDto {
  backdrop_path: string;
  created_by: TmdbCreatorDto[];
  episode_run_time: number[];
  first_air_date: string;
  genres: TmdbGenreDto[];
  homepage: string;
  id: number;
  in_production: boolean;
  languages: string[];
  last_air_date: string;
  last_episode_to_air: TmdbEpisodeDto;
  name: string;
  next_episode_to_air: TmdbEpisodeDto;
  networks: TmdbNetworkDto[];
  number_of_episodes: number;
  number_of_seasons: number;
  origin_country: string[];
  original_language: string;
  original_name: string;
  overview: string;
  popularity: number;
  poster_path: string;
  production_companies: TmdbProductionCompanyDto[];
  seasons: TmdbSeasonDto[];
  status: string;
  type: string;
  vote_average: number;
  vote_count: number;
}

File: src/main/java/ch/andreskonrad/torenta/tmdb/dto/TmdbSeriesDetailDto.java
package ch.andreskonrad.torenta.tmdb.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Objects;

@NoArgsConstructor
@Getter
@EqualsAndHashCode(of = "id")
@JsonIgnoreProperties(ignoreUnknown = true)
public class TmdbSeriesDetailDto {

    private String backdrop_path;
    private List<TmdbCreatorDto> created_by;
    private List<Integer> episode_run_time;
    private String first_air_date;
    private List<TmdbGenreDto> genres;
    private String homepage;
    private int id;
    private boolean in_production;
    private List<String> languages;
    private String last_air_date;
    private TmdbEpisodeDto last_episode_to_air;
    private String name;
    private TmdbEpisodeDto next_episode_to_air;
    private List<TmdbNetworkDto> networks;
    private int number_of_episodes;
    private int number_of_seasons;
    private List<String> origin_country;
    private String original_language;
    private String original_name;
    private String overview;
    private double popularity;
    private String poster_path;
    private List<TmdbProductionCompanyDto> production_companies;
    private List<TmdbSeasonDto> seasons;
    private String status;
    private String type;
    private double vote_average;
    private int vote_count;
    private String tagline;
}
